src/wallet/wallet2.cpp

```diff
@@ -3063,12 +3179,37 @@ bool wallet2::load_keys(const std::string& keys_file_name, const epee::wipeable_
     m_account.set_device(hwdev);
     LOG_PRINT_L0("Device inited...");
   }
+
+  if (r)
+  {
+    if (encrypted_secret_keys)
+    {
+      m_account.decrypt_keys(key);
+    }
+    else
+    {
+      // rewrite with encrypted keys, ignore errors
+      encrypt_keys(key);
```

Comment:
I think it should check if the encryption is enabled:

if (m_ask_password && !m_rpc)
  encrypt_keys(key);
Likewise for the subsequent decryption step:

if (m_ask_password && !m_rpc)
  decrypt_keys(key);
To reproduce this issue: 1) use the release CLI to create a wallet, and set ask-password 0, 2) open the wallet with this patch, 3) close it, and 4) try to open the same wallet file again, which fails.